libcudacxx/include/cuda/std/__cccl/builtin.h

```diff
@@ -1270,4 +1272,91 @@
 #  define _CCCL_BUILTIN_STRLEN(...) __builtin_strlen(__VA_ARGS__)
 #endif
 
+// Some compilers provide std::move/std::forward/etc as builtins
+#if defined(__cplusplus) // && !defined(__CUDA_ARCH__)
+// Bring in the feature test macros (needed for std::forward_like)
+#  if _CCCL_HAS_INCLUDE(<version>) // <version> should be the smallest include possible
+#    include <version>
+#  elif !_CCCL_COMPILER(NVRTC)
+#    include <ciso646> // otherwise go for the smallest possible header
+#  endif // !_CCCL_COMPILER(NVRTC)
+
+// Bring in the bits of the STL we need
+#  if defined(_GLIBCXX_VERSION)
+#    include <bits/move.h> // for move, forward, forward_like, and addressof
+#  elif defined(_LIBCXX_VERSION)
+#    include <__memory/addressof.h>
+#    include <__utility/as_const.h>
+#    include <__utility/forward.h>
+#    include <__utility/forward_like.h>
+#    include <__utility/move.h>
+#  elif defined(_MSVC_STL_VERSION)
```

Comment:
I believe we should not need this because we can just mark our functions with `_CCCL_INTRINSIC`